src/utils/formContext.ts

```diff
@@ -1,59 +1,101 @@
-export const getFormContext = (
-  newStep: any,
-  props: {
-    client: any;
-    formName: string;
-    formRef: React.MutableRefObject<any>;
-    formSettings: any;
-    getErrorCallback: (
-      props1?: Record<string, unknown>
-    ) => (props2?: Record<string, unknown>) => Promise<void>;
-    history: any;
-    setInlineErrors: React.Dispatch<
-      React.SetStateAction<Record<string, { message: string; index: number }>>
-    >;
-    setUserProgress: React.Dispatch<React.SetStateAction<null>>;
-    steps: any;
-    updateFieldOptions: (
-      stepData: any,
-      loadStep?: null
-    ) => (newOptions: FieldOptions) => void;
-  }
-) => ({
+export type ContextProps = {
+  client: any;
+  formName: string;
+  formRef: React.MutableRefObject<any>;
+  formSettings: any;
+  getErrorCallback: (
+    props1?: Record<string, unknown>
+  ) => (props2?: Record<string, unknown>) => Promise<void>;
+  history: any;
+  setInlineErrors: React.Dispatch<
+    React.SetStateAction<Record<string, { message: string; index: number }>>
+  >;
+  setUserProgress: React.Dispatch<React.SetStateAction<null>>;
+  steps: any;
+  updateFieldOptions: (
+    stepData: any,
+    loadStep?: null
+  ) => (newOptions: FieldOptions) => void;
+};
+
+/**
+ * Used by contextRef in <Form />, renderAt for vanillajs, and the lifecycle
+ * methods
+ *
+ * @param formUuid
+ * @returns Form context object
+ */
+export const getFormContext = (formUuid: string) => ({
   setValues,
-  setFormCompletion: (flag: boolean) =>
-    props.client.registerEvent({
-      step_key: newStep.key,
+  setFormCompletion: (flag: boolean) => {
+    const { contextProps, currentStep } =
+      getInternalState().formState[formUuid];
+    if (!contextProps) return;
+
+    return contextProps.client.registerEvent({
+      step_key: currentStep.key,
       event: 'load',
       completed: flag
-    }),
-  setOptions: props.updateFieldOptions(props.steps),
-  setProgress: (val: any) => props.setUserProgress(val),
+    });
+  },
+  setOptions: (newOptions: FieldOptions) => {
+    const contextProps = getInternalState().formState[formUuid].contextProps;
+    if (!contextProps) return;
+    return contextProps.updateFieldOptions(contextProps.steps)(newOptions);
+  },
+  setProgress: (val: any) => {
+    const contextProps = getInternalState().formState[formUuid].contextProps;
+    if (!contextProps) return;
+    return contextProps.setUserProgress(val);
+  },
   setStep: (stepKey: any) => {
-    changeStep(stepKey, newStep.key, props.steps, props.history);
+    const { contextProps, currentStep } =
+      getInternalState().formState[formUuid];
+    if (!contextProps) return;
+    changeStep(
+      stepKey,
+      currentStep.key,
+      contextProps.steps,
+      contextProps.history
+    );
   },
-  step: {
+  getAttributes: () => ({
+    stepName: getInternalState().formState[formUuid]?.currentStep?.key ?? '',
     style: {
-      backgroundColor: newStep?.default_background_color
+      backgroundColor:
```

Comment:
should we just flatten this for now? to backgroundColor instead of style: {backgroundColor}. IMO this will be refactored down the line anyways since default_background_color is being deprecated in favor of the grid styles